### DISCUSSION TITLE
CPU undervolt not working on Ryzen 9 7940HS

### DISCUSSION THREAD

**Discussion by s0nniez** [seerge/g-helper#675] [Q&A] [answered]
CPU undervolt seem to not be working on my 2023 G14 Ryzen 9 7940HS with RTX 4060. Looking at the log, the values were applied successfully. OC the gpu works, just not undervolting. I'm at -30 undervolt and see no difference.

**Comment by seerge** [comment]
@s0nniez hello, there are a bunch of users on reddit and here on github who reported that it works for them for exact same 7940HS, so i don't think it's related to your model.

If logs says that value is accepted, it went ok :) 

How do you verify? Try to run cinebench 23 with undervolt set to 0 and set to your max, and compare results :)

**Comment by envyska** [comment]
I use g14/4080 model without miniLED. When i put -5, -10 ,-15, my laptop finished timespy successfully. First time setting -20, my laptop automatically rebooted druing GPU Test 2. Ive checked monitoring graph, now i use -12 on my laptop. It seems to be working perfectly for me. HS models are weaker against undervolting than HX/H :(

Just stress it. your laptop will show you some difference. around -10~-15 is stable without voltage offset.

**Comment by s0nniez** [comment]
I tested -5, -10, -15, and -30 in timespy, all ran successfully. I monitor max temp in HWmonitor and see no difference in time spy score and max temp, that's why I am confused why it's not working for me but seems to be working for other users with the same config.

**Comment by s0nniez** [comment]
Updating the chipset made a little difference in max temp and timespy.
UV -30: Timespy cpu score 10968, max temp 92C, core 90C. Max VID 1.515
UV 0: Timespy cpu score 10685, max temp 96.4C, core 94C. Max VID 1.525
Is this how it's supposed to work?

**Reply by seerge** [reply, marked as answer]
Well, you seem to get 300 extra score, it's not that bad (+2.6%). Try to run CPU only benchmark (cinebench23), it usually shows bigger difference.

Just for the reference, I get 14200 points w/o UV, and 14700 points with -15UV (my max) in CB23. So + 3.5% more.

Don't expect miracles from that.

**Comment by s0nniez** [comment]
> Yes, G-helper uses UXTU's library for UV (it says that in readme) :) so results should be identical

**Comment by phush0** [comment]
It is not working on Razer Blade 14 with same CPU. I put it on -50 and there is no difference at all.
